tests/tensorflow_autolog/test_tensorflow2_autolog.py

```diff
@@ -535,77 +543,87 @@ def test_tf_keras_autolog_logs_to_and_deletes_temporary_directory_when_tensorboa
         assert not os.path.exists(mock_log_dir_inst.location)
 
 
-def create_tf_estimator_model(directory, export, training_steps=100, use_v1_estimator=False):
-    CSV_COLUMN_NAMES = ["SepalLength", "SepalWidth", "PetalLength", "PetalWidth", "Species"]
-
-    train = pd.read_csv(
-        os.path.join(os.path.dirname(__file__), "iris_training.csv"),
-        names=CSV_COLUMN_NAMES,
-        header=0,
+@dataclass
```

Comment:
The reason for wrapping everything inside a dataclass is because I wish to access the training data and the tensorflow model for one of the tests as I've discussed [here](https://github.com/mlflow/mlflow/issues/4828#issuecomment-929852602). Not sure if I made any mistake but [this](https://github.com/mlflow/mlflow/issues/4828#issuecomment-930269226) is what I followed for solving the issue.

Reply:
The reason for wrapping everything inside a dataclass is because I wish to access the training data and the tensorflow model for one of the tests as I've discussed [here](https://github.com/mlflow/mlflow/issues/4828#issuecomment-929852602). Not sure if I made any mistake in following the instructions but [this](https://github.com/mlflow/mlflow/issues/4828#issuecomment-930269226) is what I followed.

Reply:
Ah, yes that's right :) Makes total sense! It's definitely far simpler than handling additional disparate function calls.